### PR TITLE
Ports Polaris's port of salvageable machines and computers

### PR DESCRIPTION
**Pull request by CaraS212**
Adds busted down machines and computers that can be crowbar'd apart for scrap.

I need feedback on what these things should give and how much. so far putting a burnt circuit board gives you maybe 7 or so steel. Other objects have lines saying they should be recycling into gold, silver, etc. while our lathes and i believe the trash crushers do not handle the recycling of advanced materials.

per the code for the trash items: list(MATERIAL_name, chance, amount)

adds the ability for lathes to recycle advanced material.

### CODE REVIEW

File: code/modules/urist/items/salvageable.dm [hunk 1]
```diff
@@ -0,0 +1,366 @@
+/obj/structure/salvageable
+	name = "broken macninery"
+	desc = "Broken beyond repair, but looks like you can still salvage something from this if you had a prying implement."
+	icon = 'icons/urist/obj/salvageable.dmi'
+	density = 1
+	anchored = 1
+	var/salvageable_parts = list()
```
Comment (scrdest) [inline code/modules/urist/items/salvageable.dm:7]:
```suggestion
	var/list/salvageable_parts = list()
```
A bit clearer this way

**Review by scrdest — APPROVED** [review]

File: code/modules/urist/items/salvageable.dm [hunk 2]
```diff
@@ -0,0 +1,366 @@
+/obj/structure/salvageable
+	name = "broken macninery"
+	desc = "Broken beyond repair, but looks like you can still salvage something from this if you had a prying implement."
+	icon = 'icons/urist/obj/salvageable.dmi'
+	density = 1
```
Comment (scrdest) [inline code/modules/urist/items/salvageable.dm:5]:
```suggestion
	density = TRUE
```
Will pacify the linter

**Comment by CaraS212** [comment]
i cant tell if the code check is an indentation error this time, but i dont see anything immediate when  i highlight the added files

File: code/modules/urist/items/salvageable.dm [hunk 3]
```diff
@@ -0,0 +1,366 @@
+/obj/structure/salvageable
+	name = "broken macninery"
+	desc = "Broken beyond repair, but looks like you can still salvage something from this if you had a prying implement."
+	icon = 'icons/urist/obj/salvageable.dmi'
+	density = TRUE
+	anchored = 1
```
Comment (scrdest) [inline code/modules/urist/items/salvageable.dm:6]:
```suggestion
	anchored = TRUE
```
That's what it's failing at; the check wording is jank

File: code/modules/urist/modules/newtrading/trade/trade_items/tools.dm [hunk 4]
```diff
@@ -29,7 +29,6 @@
 	value = 30
 //	trader_weight = 1
 
-/datum/trade_item/tool/screwdriver
```
Comment (Glloyd) [inline code/modules/urist/modules/newtrading/trade/trade_items/tools.dm:32]: this shouldn't have been removed

**Review by Glloyd — CHANGES_REQUESTED** [review]
if you remove the trader stuff from this PR, we can get this merged. That should be for a future PR regardless of whether it works or not right now because there's balance considerations with prices; setting those trade prices will affect the prices the nerva gets for trading those items through cargo. Also, this will likely conflict with #1251 if you keep the NT station changes in here.

**Comment by CaraS212** [comment]
Ill strip out the trader stuff tomorrow and open a new draft.

**Comment by CaraS212** [comment]
removed station file entirely as it has been replaced by a 2 part dmm file

File: code/modules/urist/items/salvageable.dm [hunk 5]
```diff
@@ -0,0 +1,366 @@
+/obj/structure/salvageable
+	name = "broken macninery"
```
Comment (Glloyd) [inline code/modules/urist/items/salvageable.dm:2]:
```suggestion
	name = "broken machinery"
```

File: code/modules/urist/items/salvageable.dm [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+		/obj/item/stock_parts/subspace/transmitter = 40,
+		/obj/item/stock_parts/subspace/transmitter = 40,
+		/obj/item/stock_parts/subspace/crystal = 30,
+		/obj/item/stock_parts/subspace/crystal = 30,
+		/obj/item/stock_parts/computer/network_card/wired = 20
+	)
+
+/obj/structure/salvageable/console_os
+	name = "pristine console"
+	desc = "Despite being in pristine condition this console doesn't respond to anything, but looks like you can still salvage something from this."
```
Comment (Glloyd) [inline code/modules/urist/items/salvageable.dm:339]:
```suggestion
	desc = "Despite being in pristine condition this console doesn't respond to anything, but it looks like you can still salvage something from this."
```

**Review by Glloyd — APPROVED** [review]
this'll be great for awaymaps 👍

**Pull request merged**